src/xinspect.cpp

```diff
@@ -0,0 +1,256 @@
+/************************************************************************************
+ * Copyright (c) 2023, xeus-cpp contributors                                        *
+ *                                                                                  *
+ * Distributed under the terms of the BSD 3-Clause License.                         *
+ *                                                                                  *
+ * The full license is in the file LICENSE, distributed with this software.         *
+ ************************************************************************************/
+
+#include "xinspect.hpp"
+
+#include "clang/Interpreter/CppInterOp.h"
+
+namespace xcpp
+{
+    bool node_predicate::operator()(pugi::xml_node node) const
+    {
+        return static_cast<std::string>(node.attribute("kind").value()) == kind
+               && static_cast<std::string>(node.child("name").child_value()) == child_value;
+    }
+
+    std::string class_member_predicate::get_filename(pugi::xml_node node) const
+    {
+        for (pugi::xml_node child : node.children())
+        {
+            if (static_cast<std::string>(child.attribute("kind").value()) == kind
+                && static_cast<std::string>(child.child("name").child_value()) == child_value)
+            {
+                return child.child("anchorfile").child_value();
+            }
+        }
+        return "";
+    }
+
+    bool class_member_predicate::operator()(pugi::xml_node node) const
+    {
+        auto parent = (static_cast<std::string>(node.attribute("kind").value()) == "class"
+                       || static_cast<std::string>(node.attribute("kind").value()) == "struct")
```

Comment:
These should go in a separate temporary variable.

Reply:
Tried using some more readable names here. Hope it looks better now.